lib/fetch/body.js

```diff
@@ -397,7 +398,47 @@ function bodyMixinMethods (instance) {
 
       // If mimeType’s essence is "multipart/form-data", then:
       if (/multipart\/form-data/.test(contentType)) {
-        throw new NotSupportedError('multipart/form-data not supported')
+        const headers = {}
+        for (const [key, value] of this.headers) headers[key.toLowerCase()] = value
+
+        const responseFormData = new FormData()
+
+        let busboy
+
+        try {
+          busboy = Busboy({ headers })
+        } catch (err) {
+          // Error due to headers:
+          throw Object.assign(new TypeError(), { cause: err })
+        }
+
+        busboy.on('field', (name, value) => {
+          responseFormData.append(name, value)
+        })
+        busboy.on('file', (name, value, info) => {
+          const { filename, encoding, mimeType } = info
+          const base64 = encoding.toLowerCase() === 'base64'
+          const chunks = []
+          value.on('data', (chunk) => {
+            if (base64) chunk = Buffer.from(chunk.toString(), 'base64')
```

Comment:
It is not possible to decode each chunk from `base64' because the chunk may be an incomplete sequence of `base64`, which will lead to data corruption.

Reply:
I don't understand.

Reply:
```js
const data = Buffer.from(Buffer.from('hello world').toString('base64'));

{
	const chunks = [];
	for (let offset = 0, step = 6; offset <= data.length; offset += step) {
		const chunk = data.slice(offset, offset + step);
		chunks.push(Buffer.from(chunk.toString(), 'base64'));
	}

	console.log(await new Blob(chunks).text());
	// hell�v�ld
}

{
	const chunks = [];
	for (let offset = 0, step = 6; offset <= data.length; offset += step) {
		const chunk = data.slice(offset, offset + step);
		chunks.push(chunk);
	}

	const buffer = Buffer.from(Buffer.concat(chunks).toString(), 'base64');

	console.log(await new Blob([buffer]).text());
	// hello world
}
```

Reply:
We have the https://nodejs.org/api/string_decoder.html for this purpose.

Reply:
@mcollina does string decoder need to be applied somewhere in this code explicitly, or it will be called down the line automatically?

Reply:
I think in this case we should be using it manually.

Reply:
@cameron-robey string decoder needs to be used here, based on the above

Reply:
https://nodejs.org/api/string_decoder.html this is not what is needed. Because it decodes from buffer to a base64 string. In our case, it is supposed to decode base64 string into a buffer.

Reply:
@repsac-by any alternate suggestions?

Reply:
I haven't found any helpers without using external dependencies. We can collect all the chunks and decode them at the end, but this will entail double memory consumption. The second approach is as follows:

```js
busboy.on('file', (name, value, info) => {
          const { filename, encoding, mimeType } = info
          const base64 = encoding.toLowerCase() === 'base64'
          const chunks = []
          let base64buf = ''
          value.on('data', (chunk) => {
            if (base64) {
                base64buf += chunk.toString();
                const end = base64buf.length - base64buf.length % 4;
                chunk = Buffer.from(base64buf.slice(0, end), 'base64')
                base64buf = base64buf.slice(end);
            }
            chunks.push(chunk)
          })
          value.on('end', () => {
            if (base64) chunks.push(Buffer.from(base64buf, 'base64'))
            const file = new File(chunks, filename, { type: mimeType })
            responseFormData.append(name, file)
          })
```

Reply:
@repsac-by what external dependencies would help?

Reply:
@kibertoad https://github.com/mazira/base64-stream

Reply:
Considering that the library wasn't updated for years, and decoder is [really small](https://github.com/mazira/base64-stream/blob/master/lib/decode.js), wonder if we should just inline it in the repo.
@mcollina thoughts?

Reply:
@repsac-by btw, am I missing something, or https://github.com/mazira/base64-stream/blob/master/lib/decode.js does for the most part exact same thing as this PR?

Reply:
@kibertoad  it does what I suggested in the second case, only with the difference that it applies to [stream.Transform](https://nodejs.org/dist/latest-v18.x/docs/api/stream.html#implementing-a-transform-stream). This is convenient because `value` in our case is a stream

Reply:
I think this is the only comment left to resolve.

Reply:
What exactly needs to be implemented - the thread is unclear @mcollina

Reply:
@cameron-robey see what @repsac-by is suggesting, plus test to cover that case. inlining the code should be fine

Reply:
What's the problem exactly? You need 4 bytes to decode base64 at minimum, so why don't you accumulate until you have 4 bytes of multiples?

Reply:
@mcollina I'm assuming that you mean: push chunks until the total length of unprocessed data is divisible by 4, then decode and clear the chunk array, and repeat with the next pieces of data. Accumulating like that seems unreliable, since it cannot be known when that multiple of 4 will be reached. It might happen only after a huge number of chunks. [The streaming method suggested above](https://github.com/nodejs/undici/pull/1606#discussion_r967644495) is superior because the number of unprocessed bytes that are kept in memory never exceeds 3.

Reply:
The one in the comment also looks good.

Reply:
@cameron-robey Do you need any help with the remaining part?